mate-panel/libpanel-util/panel-keyfile.c

```diff
@@ -128,7 +128,7 @@ panel_key_file_to_file (GKeyFile     *keyfile,
 		new_length = length + strlen (KEYFILE_TRUSTED_SHEBANG);
 		new_data = g_malloc (new_length);
 
-		strcpy (new_data, KEYFILE_TRUSTED_SHEBANG);
+		g_strlcpy (new_data, KEYFILE_TRUSTED_SHEBANG, (new_length + 1));
```

Comment:
You allocate `new_length` bytes and pass `new_length + 1` as `dest_size`...

Reply:
```patch
diff --git a/mate-panel/libpanel-util/panel-keyfile.c b/mate-panel/libpanel-util/panel-keyfile.c
index ebd3249..5cf4996 100644
--- a/mate-panel/libpanel-util/panel-keyfile.c
+++ b/mate-panel/libpanel-util/panel-keyfile.c
@@ -126,11 +126,10 @@ panel_key_file_to_file (GKeyFile     *keyfile,
                gsize  new_length;
 
                new_length = length + strlen (KEYFILE_TRUSTED_SHEBANG);
-               new_data = g_malloc (new_length);
+               new_data = g_malloc (new_length + 1);
 
-               strcpy (new_data, KEYFILE_TRUSTED_SHEBANG);
-               memcpy (new_data + strlen (KEYFILE_TRUSTED_SHEBANG),
-                       data, length);
+               g_strlcpy (new_data, KEYFILE_TRUSTED_SHEBANG, new_length + 1);
+               g_strlcat (new_data, data, new_length + 1);
 
                g_free (data);
                data = new_data;
```

Reply:
`g_strdup_printf ("%s%s", KEYFILE_TRUSTED_SHEBANG, data);`

Or even better it has "//FIXME: kill this when bug #309224 is fixed" and that bug has been fixed:
https://bugzilla.gnome.org/show_bug.cgi?id=309224

Reply:
That report shows the bug was fixed back in 2013, meaning it is very unlikely any distro would ship a glib version predating the fix and also ship MATE 1.20 or 1.22. Thus we should be able to to (as the FIXME states) remove the entire function.

Reply:
@rbuj ok, done!

@lukefromdc please test again